src/v/kafka/server/replicated_partition.cc

```diff
@@ -339,6 +339,11 @@ replicated_partition::get_leader_epoch_last_offset_unbounded(
     const model::term_id term(epoch);
     const auto first_local_offset = _partition->raft_start_offset();
     const auto first_local_term = _partition->get_term(first_local_offset);
+    const auto last_local_term = _partition->term();
+    if (term > last_local_term) {
```

Comment:
LGTM but i'm curious if we've hit this in test, or if this is based on the kafka implementation? When might this actually happen? Perhaps it's more likely with follower fetching?

Reply:
Not sure maybe @VladLazar has more input?

Reply:
I don't expect the client to ever send such a request really, but I think the handling is consistent with the request semantics.